### ISSUE TITLE
Subsequent delta updates on Windows fail

### ISSUE THREAD

**Issue by mherrmann** [brave/brave-browser#29469]
## Description 

On Windows, a delta update that gets applied after another delta update always fails and falls back to the full update.

## Steps to Reproduce

   1. Install Brave 1.49.120 using its [offline installer](https://github.com/brave/brave-browser/releases/download/v1.49.120/BraveBrowserStandaloneSetup.exe). Accept the UAC prompt to perform a system-wide installation.
   2. Close Brave and make sure you don't have internet so Brave doesn't accidentally get updated.
   3. Open a Windows command prompt with Administrator privileges (right-click->"Run as Administrator").
   4. Download the [delta installer to go from Brave 1.49.120 to 1.49.128](https://updates-cdn.bravesoftware.com/delta/Brave-Release/x64-rel/win/111.1.49.128/111.1.49.120/brave_installer-delta-x64.exe).
   5. Enter the following in the command prompt: `set BraveSoftwareUpdateIsMachine=1`.
   6. Execute the delta installer in the same command prompt. For example by typing: `C:\Users\micha\Downloads\brave_installer-delta-x64.exe`. If you are testing Release, don't append anything. If you are testing Nightly, append `--chrome-sxs`. If you are testing Beta, append `--chrome-beta`. For example: `brave_installer-delta-x64.exe --chrome-beta`.
   7. Repeat steps 4. - 6. for the [delta installer to go from Brave 1.49.128 to 1.49.132](https://updates-cdn.bravesoftware.com/delta/Brave-Release/x64-rel/win/111.1.49.132/111.1.49.128/brave_installer-delta-x64.exe)
   8. Start Brave.

## Actual result:

The version on `brave://settings/help` is 1.49.128.

## Expected result:

The version on `brave://settings/help` should be 1.49.132.

## Reproduces how often: 

Always.

## Steps to Reproduce more generally

Repeat the steps above for another first, second and third _public_ versions of Brave. You can construct the delta URLs from the version numbers. Note: Deltas are only available for subsequent public versions. For example, there is also a 1.49.131 non-public/pre-release version. But because it isn't public, there is no delta for it.

## Related issues

The following issue also affects delta updates on Windows: #28989

**Comment by mherrmann** [comment]
Reopening this because my fix in #17882 only fixed one cause of the problem. But at least one other cause still exists.

**Comment by stephendonner** [comment]
> Reopening this because my fix in #17882 only fixed one cause of the problem. But at least one other cause still exists.

That's a different issue, I think you meant to point to the full link of the commit.

**Comment by mherrmann** [comment]
Thank you @stephendonner, yes, I meant https://github.com/brave/brave-core/pull/17882.

**Comment by MadhaviSeelam** [comment]
Added OS/macOS label as per the [testplan ](https://github.com/brave/brave-core/pull/18043#issue-1666327344), Widevine functionality needs to be tested on macOS. 


more info: https://bravesoftware.slack.com/archives/C012NN4F6UU/p1683648954812479

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

Brave | 1.52.94 Chromium: 113.0.5672.92 (Official Build) beta (x86_64)
-- | --
Revision | b6f521170062a1fa8a82c33fb223b06fec566da1-refs/branch-heads/5672_63@{#10}
OS | macOS Version 11.7.6 (Build 20G1231)

### Netflix

### Amazon Prime

### YouTube

**Comment by mherrmann** [comment]
I believe the milestone for this should be 1.53.x and have changed it from 1.52.x. There were two PRs that fix this issue: [17882](https://github.com/brave/brave-core/pull/17882) is in 1.52.x while [18043](https://github.com/brave/brave-core/pull/18043) is in 1.53.x. I believe this can only be fully tested with 1.53.x. What do you think @stephendonner @MadhaviSeelam?

**Comment by stephendonner** [comment]
Makes perfect sense to me, @mherrmann 👍

**Comment by kjozwiak** [comment]
> I believe the milestone for this should be 1.53.x and have changed it from 1.52.x. There were two PRs that fix this issue: [17882](https://github.com/brave/brave-core/pull/17882) is in 1.52.x while [18043](https://github.com/brave/brave-core/pull/18043) is in 1.53.x. I believe this can only be fully tested with 1.53.x. What do you think @stephendonner @MadhaviSeelam?

@mherrmann we usually keep issues in the milestone were the PRs landed. Ideally, we would have two issues for the above scenario. Once via `1.52.x` & `1.53.x`. The one in `1.52.x` would be labelled as `QA/No` as the above can only be verified via `1.53.x`. Then another issue within `1.53.x` labelled as `QA/Yes`. The rule that we usually follow is one PR per issue so you don't run into cases where one PR spans ~2-3 milestones and there's only one issue.

For now, leaving this in `1.53.x` works 👍 Regarding `1.52.x`, is there anything that needs to be checked? Just double checking that we're not going to ship with partial fixes/code that might cause problems.

**Comment by LaurenWags** [comment]
Removing `QA Pass-macOS` as milestone for this issue was adjusted.

**Comment by mherrmann** [comment]
@kjozwiak

> @mherrmann we usually keep issues in the milestone were the PRs landed. Ideally, we would have two issues for the above scenario. 

I see, that makes sense. I'll remember it for the future.

> Regarding 1.52.x, is there anything that needs to be checked?

The change in 17882 can affect Brave's Windows installer and delta updates. I presume the Windows installer gets tested as part of all other tests. For deltas, it would be good to follow the test plan in [PR #13563 ](https://github.com/brave/brave-core/pull/13563) for x64 Brave. (I see how multiple PRs for one issue make this a bit of a mess now, sorry.)

**Comment by kjozwiak** [comment]
> @kjozwiak
> 
> > @mherrmann we usually keep issues in the milestone were the PRs landed. Ideally, we would have two issues for the above scenario.
> 
> I see, that makes sense. I'll remember it for the future.
> 
> > Regarding 1.52.x, is there anything that needs to be checked?
> 
> The change in 17882 can affect Brave's Windows installer and delta updates. I presume the Windows installer gets tested as part of all other tests. For deltas, it would be good to follow the test plan in [PR #13563 ](https://github.com/brave/brave-core/pull/13563) for x64 Brave. (I see how multiple PRs for one issue make this a bit of a mess now, sorry.)

no worries! @mherrmann maybe it makes sense to open an issue via `1.52.x` just to mention that the above should be double checked? This way QA can keep an eye out. If there's any issues, we'll know why as we'll have an associated issue within `1.52.x`.

**Comment by mherrmann** [comment]
@kjozwiak I've created https://github.com/brave/brave-browser/issues/30485 now to capture this :)

**Comment by kjozwiak** [comment]
> @kjozwiak I've created #30485 now to capture this :)

awesome, thanks @mherrmann 👍 Assuming https://github.com/brave/brave-browser/issues/30551 doesn't block `1.52.x`?

**Comment by mherrmann** [comment]
> Assuming https://github.com/brave/brave-browser/issues/30551 doesn't block 1.52.x?

No, it doesn't. Not in theory and also not in practice - I just checked from v1.52.109 and v1.52.102 and successfully received updates to v1.53.66.

**Comment by MadhaviSeelam** [comment]
Verification `In progress` using






Pre-requisites: 

- Uninstall Brave from your system (all channels)
- Run the following in an Administrator command prompt: "C:\Program Files (x86)\BraveSoftware\Update\BraveUpdate.exe" /uninstall
- The following registry key should now not exist: HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\Update\Clients.
- Delete the following file if it exists: C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log.
- Set the registry value HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\BraveSoftware\UpdateDev\IsEnabledLogToFile to DWORD 1.

Upgrade `1.53.66` --> `1.53.76`
Steps: 

1. Followed above Pre-requisite steps before installing Brave
2. Installed `BraveBrowserStandaloneBetaSetup.exe` public beta installer for `1.53.66`
3. launched Brave
4. Open `brave://settings/help`
5. click `Relaunch`
6. confirmed Brave updated to latest public beta version `1.53.76`

```
[06/05/23 13:23:36.687][BraveUpdate:goopdate][15932:27616][Installing][display name: Brave Beta][app id: {103BD053-949B-43A8-9120-2E424887DE11}][installer path: C:\Program Files (x86)\BraveSoftware\Update\Install\{2FF268EE-DBD8-41A5-A307-2384995FE167}\brave_installer-delta-x64.exe][manifest args: --chrome-beta][installer data: ][untrusted data: ][priority: 10]
[06/05/23 13:23:36.687][BraveUpdate:goopdate][15932:27616][Running installer][C:\Program Files (x86)\BraveSoftware\Update\Install\{2FF268EE-DBD8-41A5-A307-2384995FE167}\brave_installer-delta-x64.exe][--chrome-beta][{103BD053-949B-43A8-9120-2E424887DE11}]
[06/05/23 13:23:55.943][BraveUpdate:goopdate][15932:27616][InstallerResult][{103BD053-949B-43A8-9120-2E424887DE11}][0]
[06/05/23 13:23:55.943][BraveUpdate:goopdate][15932:27616][InstallApp returned][0x0][{103BD053-949B-43A8-9120-2E424887DE11}][type:1][code: 30][][]
[06/05/23 13:23:55.953][BraveUpdate:goopdate][15932:27616][Started process][30708]
[06/05/23 13:23:55.962][BraveUpdate:goopdate][15932:27616][Started process][28944]
```


1.53.66 | 1.53.66 Installed logs | Update| 1.53.76 | Delta logs
------ | ------ | ------ | ------ | ------
![image](https://github.com/brave/brave-browser/assets/98358127/fa134be6-7bb3-4ce7-a608-d9007c85d813)|![image](https://github.com/brave/brave-browser/assets/98358127/51eb81bd-2bf5-4dd6-b1a3-453d4a621730)|![image](https://github.com/brave/brave-browser/assets/98358127/b2e0471a-67e1-47b2-89d7-b707d05bf10a)|![image](https://github.com/brave/brave-browser/assets/98358127/b8d0a1bd-7b6d-4f7a-b9a3-45091ee9387c)|![image](https://github.com/brave/brave-browser/assets/98358127/fc701a4d-1203-4a6e-9b41-70c21e7b7db5)

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

Brave | 1.53.83 Chromium: 114.0.5735.110 (Official Build) beta (64-bit)
-- | --
Revision | 1c828682b85bbc70230a48f5e345489ec447373e-refs/branch-heads/5735_90@{#13}
OS | Windows 10 Version 22H2 (Build 19045.3031)

**Comment by MadhaviSeelam** [comment]
@mherrmann can you review actual results section for `Re- verified as following for 1.53.76 -->1.53.82-->1.53.83` case in https://github.com/brave/brave-browser/issues/29469#issuecomment-1577500329 please? Delta updates are failing on second delta installer.  

cc: @stephendonner

**Comment by stephendonner** [comment]
> @mherrmann can you review actual results section for `Re- verified as following for 1.53.76 -->1.53.82-->1.53.83` case in [#29469 (comment)](https://github.com/brave/brave-browser/issues/29469#issuecomment-1577500329) please? Delta updates are failing (step 8) on second delta installer.
> 
> cc: @stephendonner

Perhaps (probably) due to https://github.com/brave/brave-browser/issues/30748#issue-1736714115?

**Comment by mherrmann** [comment]
Yes, unfortunately there was the second regression mentioned in @stephendonner; I'm sorry.

**Comment by stephendonner** [comment]
> Yes, unfortunately there was the second regression mentioned in #30748 by @stephendonner; I'm sorry.

No worries, and thanks for confirming.  Marking as `QA Pass-Win64` for the initial, passing case, and noted https://github.com/brave/brave-browser/issues/30748 as the followup issue 👍  cc @MadhaviSeelam

**Comment by LaurenWags** [comment]
@mherrmann we're still ok for 1.53.x without the fix to https://github.com/brave/brave-browser/issues/30748, correct? it seems that users still get updated, they just get the full installer?

cc @kjozwiak @rebron

**Comment by mherrmann** [comment]
@LaurenWags Correct!

**Comment by LaurenWags** [comment]
Excellent, thanks for confirming @mherrmann 👍🏻

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.53.91 Chromium: 114.0.5735.110 (Official Build) beta (x86_64)
-- | --
Revision | 1c828682b85bbc70230a48f5e345489ec447373e-refs/branch-heads/5735_90@{#13}
OS | macOS Version 13.4 (Build 22F66)
```

Performing Widevine checks as per https://github.com/brave/brave-core/pull/18043#issue-1666327344.

<details>
<summary>Netflix - InProgress</summary>



</details>

**Comment by LaurenWags** [comment]
Changing to `release-notes/exclude` as there additional fixes (https://github.com/brave/brave-browser/issues/30748) coming in subsequent version. Will include in release notes when all fixes are included.

**Comment by kjozwiak** [comment]
> Changing to `release-notes/exclude` as there additional fixes (#30748) coming in subsequent version. Will include in release notes when all fixes are included.
> 
> @rebron @kjozwiak @mherrmann ok with this?

SGTM 👍 No objections re: excluding the above till we have everything fixed/addressed.
